ClavaLaraApi/src-lara-clava/clava/clava/code/DoToWhileStmt.js

```diff
@@ -0,0 +1,11 @@
+laraImport("clava.ClavaJoinPoints");
+
+function DoToWhileStmt($doStmt) {
+  // do statements have an unconditional first iteration
+  const firstIterStmts = $doStmt.scopeNodes.map(($stmt) => $stmt.copy());
+
+  $doStmt.insertBefore(ClavaJoinPoints.scope(firstIterStmts));
```

Comment:
Is this copy necessary? If you just do `ClavaJoinPoints.scope($doStmt.scopeNodes) don't the nodes get automatically copied?